test/test_type.jl

```diff
@@ -10,88 +10,289 @@ outdir = "out"
 isdir(outdir) && rm(outdir, recursive = true)
 
 # Run unit tests for various equations
-@testset "Float32 Type Stability" begin
+@testset "Test Type Stability" begin
     @timed_testset "Acoustic Perturbation 2D" begin
-        v_mean_global = (0.0f0, 0.0f0)
-        c_mean_global = 1.0f0
-        rho_mean_global = 1.0f0
-        equations = AcousticPerturbationEquations2D(v_mean_global, c_mean_global,
-                                                    rho_mean_global)
-
-        x = SVector(0.0f0, 0.0f0)
-        t = 0.0f0
-        u = u_ll = u_rr = SVector(1.0f0, 1.0f0, 1.0f0, 1.0f0, 1.0f0, 1.0f0, 1.0f0)
-        orientations = [1, 2]
-        normal_direction = SVector(1.0f0, 1.0f0)
-
-        @test eltype(initial_condition_constant(x, t, equations)) == Float32
-        @test eltype(initial_condition_convergence_test(x, t, equations)) == Float32
-        @test eltype(initial_condition_gauss(x, t, equations)) == Float32
-
-        @test eltype(source_terms_convergence_test(u, x, t, equations)) == Float32
-        @test eltype(source_terms_convergence_test(u, x, t, equations)) == Float32
-
-        for orientation in orientations
-            @test eltype(flux(u, orientation, equations)) == Float32
-        end
+        for RealT in (Float32, Float64)
+            v_mean_global = (zero(RealT), zero(RealT))
+            c_mean_global = one(RealT)
+            rho_mean_global = one(RealT)
+            equations = AcousticPerturbationEquations2D(v_mean_global, c_mean_global,
+                                                        rho_mean_global)
```

Comment:
Since we're specifically interested in types here, could we please add `@inferred` tests for type stability to this file?

```suggestion
            equations = @inferred AcousticPerturbationEquations2D(v_mean_global, 
                                                                  c_mean_global,
                                                                  rho_mean_global)
```

Reply:
Make sense